crates/fuzzing/src/generators/single_inst_module.rs

```diff
@@ -0,0 +1,307 @@
+//! Generate Wasm modules that contain a single instruction.
+
+use arbitrary::{Arbitrary, Unstructured};
+use wasm_encoder::{
+    CodeSection, ExportKind, ExportSection, Function, FunctionSection, Instruction, Module,
+    TypeSection, ValType,
+};
+
+/// The name of the function generated by this module.
+const FUNCTION_NAME: &'static str = "test";
+
+/// Configure a single instruction module.
+///
+/// By explicitly defining the parameter and result types (versus generating the
+/// module directly), we can more easily generate values of the right type.
+#[derive(Clone, Debug)]
+pub struct SingleInstModule<'a> {
+    instruction: Instruction<'a>,
+    parameters: &'a [ValType],
+    results: &'a [ValType],
+}
+
+impl<'a> SingleInstModule<'a> {
+    /// Generate a binary Wasm module with a single exported function, `test`,
+    /// that executes the single instruction.
+    pub fn encode(&self) -> Vec<u8> {
+        let mut module = Module::new();
+
+        // Encode the type section.
+        let mut types = TypeSection::new();
+        types.function(
+            self.parameters.iter().cloned(),
+            self.results.iter().cloned(),
+        );
+        module.section(&types);
+
+        // Encode the function section.
+        let mut functions = FunctionSection::new();
+        let type_index = 0;
+        functions.function(type_index);
+        module.section(&functions);
+
+        // Encode the export section.
+        let mut exports = ExportSection::new();
+        exports.export(FUNCTION_NAME, ExportKind::Func, 0);
+        module.section(&exports);
+
+        // Encode the code section.
+        let mut codes = CodeSection::new();
+        let locals = vec![];
+        let mut f = Function::new(locals);
```

Comment:
This hardly matters but I'll mention it anyway: `locals` doesn't need to be a `Vec`. Since Rust 1.53 there's an `impl<T, const N: usize> IntoIterator for [T; N]`, so you can just call `Function::new([])`.